### PR TITLE
Operation branch support

### PR DESCRIPTION
**Pull request by Chris7**
Adds in support for branches and moving between branches. Not in yet is applying changes between branches.

### CODE REVIEW

File: migrations/operations/01-initial/up.sql [hunk 1]
```diff
@@ -26,3 +28,12 @@ CREATE TABLE operation_summary (
   summary TEXT NOT NULL,
   FOREIGN KEY(operation_id) REFERENCES operation(id)
 ) STRICT;
+
+CREATE TABLE branch (
+  id INTEGER PRIMARY KEY NOT NULL,
+  db_uuid TEXT NOT NULL,
+  name TEXT NOT NULL,
+  start_operation_id INTEGER,
```
Comment (dkhofer) [inline migrations/operations/01-initial/up.sql:36]: These are foreign keys on operation.id, right?  If so please add constraints

File: src/models/operations.rs [hunk 2]
```diff
@@ -261,18 +365,77 @@ impl OperationState {
...
+            .unwrap_or_else(|| panic!("No branch named {branch_name}."));
+        let mut stmt = conn
+            .prepare(
+                "INSERT INTO operation_state (db_uuid, branch_id)
+          VALUES (?1, ?2)
+          ON CONFLICT (db_uuid) DO
+          UPDATE SET branch_id=excluded.branch_id;",
+            )
+            .unwrap();
+        println!("setting branc to {branch_name}");
```
Comment (dkhofer) [inline src/models/operations.rs:399]: Typo: branc -> branch

**Comment by dkhofer** [comment]
LGTM, comments are non-blocking

**Review by dkhofer — APPROVED** [review]

**Comment by Chris7** [comment]
Thanks, ill resolve comments in the apply branch.

**Pull request merged**